main.rb

```diff
@@ -29,17 +30,61 @@ def choose_number(choice, app)
 end
 
 def main
-  app = App.new
+  # load the data and then parse it to the app
 
+  books = load_book_data
+  persons = load_persons_data
+  rentals = load_rental_data(persons, books)
+  app = App.new(books, persons, rentals)
   loop do
     render_choices
     choice = gets.chomp.to_i
     if choice >= 7
+      # we will save the data
+      app.store_data
       puts 'Thank you for using this app'
       break
     end
     choose_number(choice, app)
   end
 end
 
+def load_book_data
+  all_book = []
+  if File.exist?('data/books.json') && !File.empty?('data/books.json')
+    book_data = JSON.parset(File.read('data/books.json'))
+    book_data.each { |b| all_book.push(Book.new(b['title'], b['author'])) }
+  end
+  all_book
+end
+
+def load_persons_data
+  all_persons = []
+
+  if File.exist?('data/persons.json') && !File.empty?('data/persons.json')
+    all_persons_data = JSON.parse(File.read('data/persons.json'))
+
+    all_persons = all_persons_data.map do |_persons_data|
+      if person_data['type'] == 'student'
+        Student.new(person_data['age'], parent_permission: person_data['parent_permission'], name: person_data['name'])
+      else
+        Teacher.new(person_data['age'], person_data['specialization'], name: person_data['name'])
```

Comment:
- [ ] Here also, you need to meticulously pay attention, the map iteration name is different from what you used in the iteration, this `_persons_data` is what you named it, this `person_data` is what you later continued to use. Kindly be consistent to avoid errors